### PR TITLE
Slime Boss - Code

### PR DESCRIPTION
**Pull request by TideSofDarK**

### CODE REVIEW

**Comment by l34Um1** [comment]
Shake isn't supposed to make the boss invincible. It shouldn't be interruptable though. It should also instantly show where the projectiles are going to land and it should last longer with each cycle having projectiles going to a new location.

**Comment by TideSofDarK** [comment]
Please explain "last longer" part. You mean this stunned condition?

**Comment by TideSofDarK** [comment]
So it shouldnt be single burst and projectiles should appear consequentially in groups of like 6 or smth, right?

**Comment by l34Um1** [comment]
Yes

**Comment by TideSofDarK** [comment]
Here it is. Projectile number, delay etc, everything, can be adjusted in ability kv.

**Comment by l34Um1** [comment]
I just had the boss split after using shake even though it had taken no damage. Also, the AI is not working for me. Shake is good now though.

I'm also getting this popup. Not sure what it is.
unbenannt

**Comment by TideSofDarK** [comment]
@l34Um1 did you spawn it using console? chris asked me to make it work using bosses.lua, and that way it adds BossTier variable to boss on spawn. ai is not working if its not set
i will make a workaround to all my bosses now

**Comment by l34Um1** [comment]
Oh, I'm just spawning using cheats. Can you explain to me how to spawn them with console?

**Comment by TideSofDarK** [comment]
It should work with cheats now. 
``dota_create_unit npc_dota_boss_slime`` is console command

**Comment by carlosrpg** [comment]
@TideSofDarK 
on \game\scripts\vscripts\components\boss\spawn.lua ln 86 : 
BossSpawner:SpawnBoss(pit, bossName, bossTier, isProtected)
replace bossName for your boss name.

On \game\scripts\vscripts\settings.lua

BOSS_RESPAWN_START replace BOSS_RESPAWN_START for 0 if you want the boss to spawn immediatly.

**Comment by TideSofDarK** [comment]
Fixed both things

**Comment by l34Um1** [comment]
Please make him invincible during the shake before a split, but only during that one. Then we are done and I think this can be merged.

**Comment by TideSofDarK** [comment]
done

**Comment by carlosrpg** [comment]
@l34Um1 do you want to mege this before the model is done? if so, which model do you have in mind?
also, which tier it will be?

**Comment by l34Um1** [comment]
Yes. Isn't there a siltbreaker slime boss model?  Tier is in the issue for the bosses

**Comment by carlosrpg** [comment]
Why do the Swiper particles are in this PR? can you remove them?

**Comment by TideSofDarK** [comment]
Should be fine

**Comment by l34Um1** [comment]
Tides you gonna do the changes carlos requested?

**Comment by TideSofDarK** [comment]
Yeah sure

File: game/scripts/vscripts/abilities/slime/boss_slime_split.lua [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+------------------------------------------------------------------------------------
+
+function modifier_boss_slime_split_passive:OnDeath(keys)
+	if IsServer() then
+		local caster = self:GetParent()
+		if keys.unit:entindex() == caster:entindex() then
+			local unitName = caster:GetUnitName()
+
+			local function CreateClone(origin)
+				local clone = CreateUnitByName(unitName, origin, true, nil, nil, caster:GetTeamNumber())
```
[the remark this answers is not in the document]
Comment (TideSofDarK) [reply game/scripts/vscripts/abilities/slime/boss_slime_split.lua]: fixed

**Comment by TideSofDarK** [comment]
@l34Um1 imo ready to be merged

File: game/scripts/vscripts/units/ai_slime_spawner.lua [hunk 2]
```diff
@@ -0,0 +1,56 @@
+function Spawn( kv )
+	if not IsServer() then
+		return
+	end
+
+	if thisEntity == nil then
+		return
+  end
+  thisEntity.bForceKill = false
```
Comment (chrisinajar) [inline game/scripts/vscripts/units/ai_slime_spawner.lua]: This whitespace in this file is insane, go through and fix up the indentation. Using an editor that supports `.editorconfig` will help a lot.

**Pull request merged**